src/openloco/openloco.cpp

```diff
@@ -883,6 +895,239 @@ namespace openloco
+    static loco_global<uintptr_t[80], 0x004F9548> _4F9548;
+    static loco_global<uint8_t[80], 0x004F9688> _4F9688;
+
+    static loco_global<tile_element*, 0x009C68D0> _9C68D0;
+
+    static loco_global<coord_t, 0x009C68E4> _game_command_map_z;
+    static loco_global<string_id[1], 0x112C826> _commonFormatArgs;
+    static loco_global<uint8_t, 0x009C68EE> _errorCompanyId;
+    static loco_global<string_id, 0x009C68E8> _9C68E8;
+
+    static uint32_t loc_4314EA();
+    static uint32_t loc_4313C6(registers& regs);
+
+    static uint32_t do_gc(registers& regs)
+    {
+        uint16_t flags = regs.bx;
+
+        _gameCommandFlags = regs.bx;
+        if (game_command_nest_level != 0)
+            return loc_4313C6(regs);
+
+        console::log("Do game command %d", regs.esi);
+        if ((flags & 1) == 0)
+        {
+            return loc_4313C6(regs);
+        }
+
+        if ((flags & 0x50) != 0
+            && _4F9688[regs.esi] == 1
+            && _updating_company_id == _player_company[0])
+        {
+            if (get_pause_flags() & 1)
+            {
+                paused_state = paused_state ^ 1;
+                WindowManager::invalidate(WindowType::timeToolbar);
+                audio::unpause_sound();
+                addr<0x0050A004, uint16_t>() |= 1;
+            }
+
+            if (game_speed != 0)
+            {
+                game_speed = 0;
+                WindowManager::invalidate(WindowType::timeToolbar);
+            }
+
+            if (is_paused())
+            {
+                _9C68E6 = string_ids::empty;
+                return 0x80000000;
+            }
+        }
+
+        if (_updating_company_id == _player_company[0] && isNetworked())
+        {
+            call(0x0046E34A); // some network stuff. should have variables set but meh
+        }
+
+        return loc_4313C6(regs);
+    }
+
+    static uint32_t loc_4313C6(registers& regs)
+    {
+        uint16_t flags = regs.bx;
+        _9C68E6 = string_ids::null;
+        game_command_nest_level++;
+
+        auto addr = _4F9548[regs.esi];
+
+        uint16_t flagsBackup = _gameCommandFlags;
+        registers fnRegs = regs;
+        fnRegs.bl &= ~1;
+        console::log("addr %d (0x%X) [EAX:%X EBX:%X ECX:%X EDX:%X EDI:%X EBP:%X]", fnRegs.esi, addr, fnRegs.eax, fnRegs.ebx, fnRegs.ecx, fnRegs.edx, fnRegs.edi, fnRegs.ebp);
+        call(addr, fnRegs);
+        uint32_t ebx = fnRegs.ebx;
+        _gameCommandFlags = flagsBackup;
+
+        console::log("EBX: %X", ebx);
+        if (ebx != 0x80000000)
+        {
+            if (is_editor_mode())
+                ebx = 0;
+
+            if (game_command_nest_level == 1)
+            {
+                if ((_gameCommandFlags & 4) == 0
+                    && (_gameCommandFlags & 0x20) == 0
+                    && ebx != 0)
+                {
+                    registers regs2;
+                    regs2.ebp = ebx;
+                    call(0x0046DD06, regs2);
+                    ebx = regs2.ebp;
```

Comment:
It's a little confusing that you're updating a variable named `ebx` and not `_508F00`, but it's doing the same thing, I think.